Minio/DataModel/ObjectOperationsArgs.cs

```diff
@@ -607,4 +611,141 @@ public override RestRequest BuildRequest(RestRequest request)
             return request;
         }
     }
+
+    public class SetObjectRetentionArgs : ObjectVersionArgs<SetObjectRetentionArgs>
+    {
+        internal bool BypassGovernanceMode { get; set; }
+        internal RetentionMode Mode { get; set; }
+        internal DateTime RetentionUntilDate { get; set; }
+        internal int RetentionValidDays { get; set; }
+        public SetObjectRetentionArgs()
+        {
+            this.RequestMethod = Method.PUT;
+            this.RetentionUntilDate = default(DateTime);
+            this.RetentionValidDays = -1;
+            this.Mode = RetentionMode.GOVERNANCE;
+        }
+
+        public override void Validate()
+        {
+            base.Validate();
+            if (this.RetentionUntilDate.Equals(default(DateTime)) && this.RetentionValidDays < 0)
+            {
+                throw new InvalidOperationException("Retention Period is not set. Please set " +
+                        nameof(RetentionUntilDate) + " or " + nameof(RetentionValidDays));
+            }
+            if (!this.RetentionUntilDate.Equals(default(DateTime)) && (this.RetentionUntilDate - DateTime.Today).TotalDays <= 0.0)
```

Comment:
This check is wrong. Please remove the days argument. Not sure why this was added here.

Reply:
Days as parameter removed. Only date is retained